backend/apps/ifc_validation/checks/header_policy/validate_header.py

```diff
@@ -128,19 +165,69 @@ def populate_header(cls, values):
-        # https://github.com/buildingSMART/IFC4.x-IF/tree/header-policy/docs/IFC-file-header#description
-        v = v[0].replace(' ', '') # allow whitespaces
-        allowed_descriptions = yaml.safe_load(open(os.path.join(os.path.dirname(__file__), 'valid_descriptions.yaml')))
-        
-        schema_identifier = values.data.get('file').schema_identifier
+        """
+        https://github.com/buildingSMART/IFC4.x-IF/tree/header-policy/docs/IFC-file-header#description
+        For grammar refer to https://standards.buildingsmart.org/documents/Implementation/ImplementationGuide_IFCHeaderData_Version_1.0.2.pdf
+        """
+        header_description_text = ' '.join(v)
         try:
-            if not v in list(map(lambda desc: desc.replace(" ", ""), allowed_descriptions.get(schema_identifier))):
-                values.data.get('validation_errors').append(values.field_name)
-            if not v: # description field is mandatory
-                values.data.get('validation_errors').append(values.field_name)
+            parsed_description = parse_mvd(header_description_text)
         except:
-            pass  ## ---> what to do with invalid schema identifier? (IFC4X3)
-        return (v,)
+            pass
+        view_definitions = parsed_description.mvd
+        values.data['mvd'] = view_definitions
+
+        
+        # comments is a free textfield, but constrainted to 256 characters
+        if len(parsed_description.comments) > 256:
+            values.data.get('validation_errors').append(values.field_name)
+            return v if type(v) == tuple else v
+
+        
+        # AddonViews cannot be used without CoordinationView.
+        view_definitions_lowercase = {mvd.lower() for mvd in view_definitions}
+        if any('addonview' in mvd for mvd in view_definitions_lowercase) and 'coordinationview' not in view_definitions_lowercase:
+            values.data.get('validation_errors').append(values.field_name)
+            return v if type(v) == tuple else v
+
+            
+        # anything besides exactly CoordinationView as a single item gets excluded
+        if len(view_definitions) == 1 and view_definitions[0].lower() != 'coordinationview':
+            values.data.get('validation_errors').append(values.field_name)
+            return v if type(v) == tuple else v
+
+        
+        return v if type(v) == tuple else v
+
+    @field_validator('mvd', mode='after')
+    def validate_and_set_mvd(cls, v, values):
+        """
+        This function runs after the other fields. It validates the mvd based on the grammar done in  
+        the 'description' field. The function checks the constraints on the mvds and returns 
+        the comma separated list into a single string.
+        """
+        view_definitions = values.data.get('mvd')
+        if not view_definitions:
+            values.data.get('validation_errors').append(values.field_name)
+            return ', '.join(view_definitions)
```

Comment:
This either returns `''` or throws an exception because we have asserted in the if statement that `view_definitions` is `None/False/''/[]`.